### PR TITLE
Run Build and Validate against java 17

### PR DESCRIPTION
**Pull request by tanvigour**
Signed-off-by: tanvigour <tanvi.gour@gmail.com>

# Description

_Please explain the changes you've made_

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #701

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

**Review by mukundansundar — COMMENTED** [review]
Can we remove 13, 15 ,16 ? They are out of support ... 
cc @artursouza thoughts ?

**Comment by mukundansundar** [comment]
@tanvigour I think we need to probably move to j11 and also might need to update some of the dependencies for the build to succeed in j17 ...

**Review by artursouza — CHANGES_REQUESTED** [review]
FindBugs is failing. You might need to update this dependency too.

**Comment by mukundansundar** [comment]
> > Can we remove 13, 15 ,16 ? They are out of support ... cc @artursouza thoughts ?
> 
> Let's keep those for now since some devs might still be using them, also there is not really any cost in running those.

The cost is there is still some flakiness in our tests and when running them in parallel across different version, sometimes they fail in one of them and causes all the other executions to cancel.

File: pom.xml [hunk 1]
```diff
@@ -26,7 +26,7 @@
     <maven.deploy.skip>true</maven.deploy.skip>
 
     <gpg.skip>true</gpg.skip>
-    <spotbugs.version>4.0.0-RC1</spotbugs.version>
+    <spotbugs.version>4.6.0</spotbugs.version>
     <spotbugs.fail>true</spotbugs.fail>
```
Comment (tanvigour) [inline pom.xml:30]: @mukundansundar , @artursouza : It looks like we have enabled spotbugs to fail on error and interrupt the build. Do we want it to block the build?  In newer version it has converted all the warnings to errors.

Comment (artursouza) [reply pom.xml:30]: If we upgrade, then we need to fix the spotbugs errors.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -18,7 +18,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        java: [ 11, 13, 15, 16 ]
```
Comment (artursouza) [inline .github/workflows/build.yml:21]: Please, keep 13 and 15 since we are not winning anything by removing those.

Comment (mukundansundar) [reply .github/workflows/build.yml:21]: We are ... in the sense we are removing some flakiness in the test suite ... not necessarily the tests themselves ...

**Pull request closed without merging**